optuna/trial/_frozen.py

```diff
@@ -317,8 +317,11 @@ def set_system_attr(self, key: str, value: Any) -> None:
 
     def _validate(self) -> None:
 
-        if self.datetime_start is None:
-            raise ValueError("`datetime_start` is supposed to be set.")
+        if self.state != TrialState.WAITING:
+            if self.datetime_start is None:
+                raise ValueError(
+                    "`datetime_start` is supposed to be set when the trial state is not waiting."
+                )
```

Comment:
Can you add unit tests for this logic in `test_frozen.py`?

Reply:
Added.